Lib/test/_test_embed_structseq.py

```diff
@@ -48,7 +48,14 @@ def test_sys_funcs(self):
 
 
 try:
-    unittest.main()
+    unittest.main(
+        module=(
+            '__main__'
+            if __name__ == '__main__'
+            # Avoiding a circular import:
+            else sys.modules['test._test_embed_structseq']
+        )
+    )
```

Comment:
Hmm, if `unittest.main` takes module names, would `unittest.main(__name__)` work here?

```suggestion
    # This should run as a script even if it's not `__main__`.
    unittest.main(__name__)
```

Reply:
No: 

```
» ./python.exe -m test test_embed -m test_presite           
Using random seed: 989874951
0:00:00 load avg: 2.36 Run 1 test sequentially in a single process
0:00:00 load avg: 2.36 [1/1] test_embed
test test_embed failed -- Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython2/Lib/test/test_embed.py", line 1987, in test_presite
    self.assertIn("Tests passed", out)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
AssertionError: 'Tests passed' not found in 'pre-site import failed:\nTraceback (most recent call last):\n  File "/Users/sobolev/Desktop/cpython2/Lib/test/_test_embed_structseq.py", line 51, in <module>\n    unittest.main(\n    ~~~~~~~~~~~~~^\n        __name__\n        ^^^^^^^^\n    )\n    ^\n  File "/Users/sobolev/Desktop/cpython2/Lib/unittest/main.py", line 74, in __init__\n    self.module = getattr(self.module, part)\n                  ~~~~~~~^^^^^^^^^^^^^^^^^^^\nAttributeError: cannot access submodule \'_test_embed_structseq\' of module \'test\' (most likely due to a circular import)\nunique-python-message'

test_embed failed (1 failure)

== Tests result: FAILURE ==

1 test failed:
    test_embed

Total duration: 102 ms
Total tests: run=1 (filtered) failures=1
Total test files: run=1/1 (filtered) failed=1
Result: FAILURE
```